PickMeUp/PickMeUp/ViewControllers/DM/PMDisplayConversationsViewController.m

```diff
@@ -52,11 +58,15 @@ -(void)_runGetQuery {
     PFQuery *toQuery = [PFQuery orQueryWithSubqueries:@[user1, user2]];
     [toQuery findObjectsInBackgroundWithBlock:^(NSArray *convos, NSError *error) {
         if (convos != nil) {
-            for (PMConversation *convo in convos) {
-                [self.convoTree addConversation:convo];
+            if (convos.count > self.arrayToDisplay.count) {
+                for (int i = 0; i < convos.count-self.arrayToDisplay.count; i++) {
+                    [self.convoTree addConversation:convos[i]];
+                }
+                self.arrayToDisplay = convos;
+                [PMCachingFunctions updateCache:self.arrayToDisplay];
+                self.arrayToDisplay = [self.convoTree retreiveSubTree:kEmpt];
+                [self.tableView reloadData];
```

Comment:
BTW, make sure this code runs on the main thread. It's not clear to me from the method name which thread is used to call the completion block.